File: gcc/testsuite/gfortran.dg/gomp/adjust-args-2.f90
! Test resolution of OMP clause adjust_args
! { dg-do compile } 

module main
  implicit none
interface
subroutine f1 (i)
  integer, intent(inout) :: i
end subroutine
end interface
contains

  subroutine f3 (i)
    integer, intent(inout) :: i
    !$omp declare variant (f1) match (construct={dispatch}) adjust_args (nothing: z)  ! { dg-error "Symbol 'z' at .1. has no IMPLICIT type" }
! { dg-error "Expected dummy parameter name or a positive integer at .1." "" { target *-*-* } .-1 }
  end subroutine
  
end module

File: gcc/testsuite/gcc.dg/gomp/adjust-args-2.c
void f(int *);
#pragma omp declare variant(f) adjust_args(need_device_addr: x)
/* { dg-error "expected 'nothing' or 'need_device_ptr'" "" { target *-*-* } .-1 }  */
/* { dg-note "'need_device_addr' is not valid for C" "" { target *-*-* } .-2 }  */
void g(int *x);
